app/src/main/java/com/example/dontcallindrunk/MainActivity.kt

```diff
@@ -2,10 +2,36 @@ package com.example.dontcallindrunk
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProviders
+import androidx.room.Room
+import com.example.dontcallindrunk.data.WorkDatabase
+import com.example.dontcallindrunk.databinding.ActivityMainBinding
+import com.example.dontcallindrunk.list.ListFragment
+import com.example.dontcallindrunk.record.RecordFragment
+import com.example.dontcallindrunk.setting.SettingFragment
 
 class MainActivity : AppCompatActivity() {
+
+    private val mainActivityBinding by lazy { DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main) }
+
+    private val viewModel by lazy { ViewModelProviders.of(this@MainActivity).get(MainViewModel::class.java)}
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+
+        mainActivityBinding.viewModel = viewModel
+
+        mainActivityBinding.fragmentManager = supportFragmentManager
```

Comment:
Kotlin - apply, also / let, run의 차이. 언제 쓰는지 알아보자.

apply -> A.apply{ A = this } : A  
also -> A.also{ A = it } : A

run -> A.run { A = this, return C } : C
let -> A.let { A = it, return G} : G

```suggestion
private val mainActivityBinding : ActivityMainBinding by lazy { DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main).apply {
    this.viewModel = viewModel
} }
private val viewModel : MainViewModel by lazy { ViewModelProviders.of(this@MainActivity).get(MainViewModel::class.java)}
```

Reply:
```kotlin
mainActivityBinding.viewModel = viewModel
```
이 코드를 Binding 에 apply 로 묶어주면 viewModel 이 호출되지 않아서
```kotlin
mainActivityBinding.fragmentManager = supportFragmentManager
```
이 코드를 apply 로 묶어줌